Document distribution API (#131)
discord-userdoccers/discord-userdoccers: pages/topics/client-distribution.mdx

###### Response Body

| Name       | Type              | Description                            |
| ---------- | ----------------- | -------------------------------------- |
| name       | string            | The latest host version                |
| pub_date   | ISO8601 timestamp | When the update was published          |
| url? ^1^   | string            | The URL to the corresponding installer |
| notes? ^1^ | string            | Any extra notes for the update         |

^1^ Only provided if auto updates are available for the selected platform.

###### Example Response

```json
{
  "name": "0.0.75",
  "pub_date": "2023-07-05T17:16:10",
  "url": "https://dl-ptb.discordapp.net/apps/osx/0.0.75/DiscordPTB.zip",
  "notes": ""
}
```

<RouteHeader method="GET" url="/modules/{release_channel}/versions.json" deprecated>
  Get Native Module Versions
</RouteHeader>

Returns a mapping of module names to integer versions representing the native module versions for the provided [release channel](#desktop-release-channel) and selected platform, if found.

{/* TODO: How the fuck is this used */}

<Alert type="warn">

> For certain platforms, this endpoint is deprecated. It is in the process of being replaced by [Get Latest Distributed Application Manifest](#get-latest-distributed-application-manifest).

Reviewer: On Windows it is not "deprecated", it has been completely unused for years and older clients using it would not work at all.

Author: How do you think I can word this better?

# Client Distribution

> While mobile clients can be distributed through the Google Play Store and Apple App Store, desktop clients require a custom solution for downloading and updating. Discord offers various APIs for downloading clients, their native modules, and keeping everything up-to-date. Visit the [downloads page](https://discord.com/download) to learn more about the clients currently offered.

Reviewer 2: ```suggestion
While mobile clients can be distributed through most system app stores, desktop clients require a custom solution for downloading and updating. Discord provides various APIs for downloading clients, their native modules, and keeping everything up-to-date. Visit the [downloads page](https://discord.com/download) to learn more about the clients currently offered.
```

###### Distribution Base URLs

These URLs, provided for convenience, provide access to all client downloads (and redirect access to [CDN-hosted ones](/reference#cdn-formatting)), but are NOT used for the API requests below. For those, see the [API Base URL](/reference#base-url).

```
https://dl.discordapp.net/
https://dl-ptb.discordapp.net/
https://dl-canary.discordapp.net/
https://dl-development.discordapp.net/
```

## Clients

### Web

> Discord offers a web client that can be used in a browser. This same web client is also used in the desktop client.

Reviewer 3: > This same web client is also used in the desktop client.
Should we clarify here that the desktop client uses native modules for some things

Author: Done!

###### Web Release Channel

| Value      | URL                            | Description       |
| ---------- | ------------------------------ | ----------------- |
| stable     | https://discord.com/app        | Stable build      |

> | ptb ^1^    | https://ptb.discord.com/app    | Public test build |

Reviewer 2: ```suggestion
| ptb ^1^    | https://ptb.discord.com/app    | Public/beta test build |
```

If we’re going to add “alpha” below

Author: I don't see the need for this

| Value           | Description       |
| --------------- | ----------------- |
| stable          | Stable build      |
| ptb ^1^         | Public test build |
| canary ^1^      | Alpha test build  |
| development ^2^ | Development build |

^1^ See the [Help Center article](https://support.discord.com/hc/en-us/articles/360035675191-Discord-Testing-Clients) for more information on Discord testing clients.

^2^ The development build follows the [`canary` web release channel](#web-release-channel) and is not recommended for use. It may be unstable or broken at any time.

###### Desktop Platform Type

| Value | Description |
| ----- | ----------- |
| win   | Windows     |
| osx   | macOS       |
| linux | Linux       |

###### Desktop Architecture Type

| Value | Description  |
| ----- | ------------ |
| x86   | 32-bit build |

###### Desktop Executable Format

| Value  | Description                  |
| ------ | ---------------------------- |
| deb    | Debian software package file |
| tar.gz | Compressed archive file      |

### Mobile

> Discord offers stable and beta mobile clients for both Android and iOS.

Reviewer 2: ```suggestion
Discord maintains stable and beta mobile clients for both Android and iOS.
```

###### Android Release Channel

<Alert type="warn">

> Official Android clients employ a minimum version check at startup. If the installed version is below the minimum version, the client will refuse to start and prompt the user to update.

Reviewer 2: ```suggestion
Official Android clients include a minimum version check at startup. If the installed package’s version is below the minimum version, the client will refuse to start and will prompt the user to update.
```

Author: This just makes it flow worse.

> This minimum version check is available at `https://dl.discordapp.net/apps/android/versions.json`. The response contains a single `discord_android_min_version` field, which is a string containing the minimum version.

Reviewer 2: ```suggestion
This minimum version data is available at `https://dl.discordapp.net/apps/android/versions.json`. The response contains a `discord_android_min_version` field, which is a string containing the minimum version.
```

</Alert>

| Value     | URL                                                       | Description              |
| --------- | --------------------------------------------------------- | ------------------------ |
| stable    | https://play.google.com/store/apps/details?id=com.discord | Stable application build |
| beta ^1^  | https://play.google.com/apps/testing/com.discord          | Beta application build   |
| alpha ^1^ | https://groups.google.com/g/discord-android-alpha-testers | Alpha application build  |

^1^ See the [Help Center article](https://support.discord.com/hc/en-us/articles/360035675191-Discord-Testing-Clients) for more information on Discord testing clients.

###### iOS Release Channel

| Value     | URL                                                                  | Description                 |
| --------- | -------------------------------------------------------------------- | --------------------------- |
| stable    | https://apps.apple.com/us/app/discord-talk-chat-hang-out/id985746746 | Stable application build    |
| beta ^1^  | https://testflight.apple.com/join/gdE4pRzI                           | Beta application build      |

> | alpha ^1^ | <private\>                                                           | Internal application builds |

Reviewer 2: What’s the point of this `<private\>`

Author: There's no available link, so I just put `<private>`

Redirects to the latest application installer for the provided [release channel](#desktop-release-channel) and selected platform.

<Alert type="warn">

This endpoint is in the process of being decommissioned in favor of [Get Latest Distributed Application Installer](#get-latest-distributed-application-installer).

</Alert>

###### Query String Params

| Name        | Type   | Description                                                                                  |
| ----------- | ------ | -------------------------------------------------------------------------------------------- |
| platform    | string | The [platform](#desktop-platform-type) to get the installer for                              |
| format? ^1^ | string | The [executable format](#desktop-executable-format) to get the installer for (default `deb`) |

^1^ Only applicable to the [Linux platform](#desktop-platform-type).

<RouteHeader method="GET" url="/updates/{release_channel}" deprecated>
  Get Application Updates
</RouteHeader>

Returns information about the latest application host update for the provided [release channel](#desktop-release-channel) and selected platform.

<Alert type="warn">

> This endpoint is in the process of being decommissioned in favor of [Get Latest Distributed Application Manifest](#get-latest-distributed-application-manifest).

Reviewer 3: Possibly add a note here mentioning that linux and macOS still use it

Author: I think "in the process of" and the note on the endpoint cover that.

> This endpoint is in the process of being decommissioned in favor of [Get Latest Distributed Application Manifest](#get-latest-distributed-application-manifest).

Reviewer 2: ```suggestion
This endpoint is deprecated in favour of [Get Latest Distributed Application Manifest](#get-latest-distributed-application-manifest).
```

Author: See conversations above
